integration-tests/src/test/java/org/apache/fineract/integrationtests/datatable/DatatableIntegrationTest.java

```diff
@@ -615,13 +619,41 @@ public void validateCreateAndEditDatatable() {
         // Read the Datatable entry generated with genericResultSet in true (default)
         items = this.datatableHelper.readDatatableEntry(datatableName, clientId, genericResultSet, null, "");
         assertNotNull(items);
-        assertEquals(1, ((List) items.get("data")).size());
-        LOG.info("Record created at {}", ((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(3));
-        LOG.info("Record updated at {}", ((List) ((Map) ((List) items.get("data")).get(0)).get("row")).get(4));
+        data = (List) items.get("data");
+        assertEquals(1, data.size());
+
+        records = (List) ((Map) data.get(0)).get("row");
+        LOG.info("Record created at {}", records.get(3));
+        LOG.info("Record updated at {}", records.get(4));
+
+        assertEquals(clientId, records.get(0));
+        assertEquals(randomString, records.get(2));
+        assertEquals(textValue, records.get(5));
+
+        Integer resourceId = this.datatableHelper.deleteDatatableEntries(datatableName, clientId, "resourceId");
+        assertEquals(clientId, resourceId, "ERROR IN DELETING THE DATATABLE ENTRIES");
+
+        // Update - update, delete DataTable columns
+        columnMap = new HashMap<>();
+        columnMap.put("apptableName", CLIENT_APP_TABLE_NAME);
+        columnMap.put("entitySubType", CLIENT_PERSON_SUBTYPE_NAME);
+        List<Map<String, Object>> dropColumnsList = Collections.singletonList(Collections.singletonMap("name", "itsANumber"));
+        columnMap.put("dropColumns", dropColumnsList);
+        ArrayList<HashMap<String, Object>> changeColumnsList = new ArrayList<>();
+        DatatableHelper.addDatatableColumns(changeColumnsList, "itsAString", null, false, 100, null);
+        columnMap.put("changeColumns", changeColumnsList);
+        datatabelRequestJsonString = new Gson().toJson(columnMap);
+        LOG.info("map to update : {}", datatabelRequestJsonString);
+        datatableUpdateResponse = this.datatableHelper.updateDatatable(datatableName, datatabelRequestJsonString);
+        assertNotNull(datatableUpdateResponse);
+        assertEquals(datatableName, datatableUpdateResponse.getResourceIdentifier());
```

Comment:
Would you mind please test whether renaming a column is working as expected?

Reply:
There is a test already for rename in DatatableUniqueAndIndexColumnTest.testDatableModificationWithUniqueAndIndexedColumns